lib/determinator/tracking/rack/middleware.rb

```diff
@@ -0,0 +1,24 @@
+require 'determinator/tracking'
+
+module Determinator
+  module Tracking
+    module Rack
+      class Middleware
+        def initialize(app)
+          @app = app
+        end
+
+        def call(env)
+          Determinator::Tracking.start!(:rack)
+          status, headers, response = @app.call(env)
+          [status, headers, response]
+        rescue
```

Comment:
this will only log an "error" if the middleware is inserted before sentry/other exception tracking — what is the use case of logging "error" as a separate field from status?

Reply:
All middlewares _should_ re-raise exceptions (Sentry for instance:(https://github.com/getsentry/raven-ruby/blob/21669e97ee704efe40432ae66a1078cf74030ca6/lib/raven/integrations/rack.rb#L56)). The status will be set to 500 only later by rails at the end of the callback unwinding.

Reply:
Okay got it — is there a use case for tracking "exceptions" as a separate state from "5xx" status codes?

Reply:
It's not that - a middleware will not see the 500 status code, because that's set by Rails at the end of the chain. Here we can only track statuses explicitly returned by the app.

Reply:
I think that depends on app specific behaviour — for example in orderweb exception handling is right in the middle of the middleware stack:

![Screenshot 2020-02-05 at 11 20 02](https://user-images.githubusercontent.com/816243/73837691-e4473980-4809-11ea-8041-e7efcafcb97d.png)

with this, what we would capture would depend on if this is inserted before or after the exception handling middleware. is there any way we can make this more predictable? as otherwise we would need to "cast" error status out from the two states we record in the processing pipeline